src/auspex/analysis/qubit_fits.py

```diff
@@ -0,0 +1,375 @@
+
+import numpy as np
+from auspex.log import logger
+from copy import copy
+import matplotlib.pyplot as plt
+from .fits import AuspexFit
+from .signal_analysis import KT_estimation
+
+class RabiAmpFit(AuspexFit):
+    """A fit to a Rabi amplitude curve, assuming a cosine model.
+    """
+
+    xlabel = "Amplitude"
+    ylabel = r"<$\sigma_z$>"
+    title = "Rabi Amp Fit"
+
+    @staticmethod
+    def _model(x, *p):
+        return p[0] - p[1]*np.cos(2*np.pi*p[2]*(x - p[3]))
+
+    def _initial_guess(self):
+        #seed Rabi frequency from largest FFT component
+        N = len(self.ypts)
+        yfft = np.fft.fft(self.ypts)
+        f_max_ind = np.argmax(np.abs(yfft[1:N//2]))
+        f_0 = 0.5 * max([1, f_max_ind]) / self.xpts[-1]
+        amp_0 = 0.5*(self.ypts.max() - self.ypts.min())
+        offset_0 = np.mean(self.ypts)
+        phase_0 = 0
+        if self.ypts[N//2 - 1] > offset_0:
+            amp_0 = -amp_0
+        return [offset_0, amp_0, f_0, phase_0]
+
+    def _fit_dict(self, p):
+
+        return {"y0": p[0],
+                "Api": p[1],
+                "f": p[2],
+                "phi": p[3]}
+
+    def __str__(self):
+        return "y0 - Api*cos(2*pi*f*(t - phi))"
+
+    @property
+    def pi_amp(self):
+        """Returns the pi-pulse amplitude of the fit.
+        """
+        return self.fit_params["Api"]
+
+    def annotation(self):
+        return r"$A_\pi$ = {0:.2e} {1} {2:.2e}".format(self.fit_params["Api"], chr(177), self.fit_errors["Api"])
+
+class RabiWidthFit(AuspexFit):
+    """Fit to a single-frequency decaying cosine for fitting Rabi-vs-time experiments
+    """
+    xlabel = "Delay"
+    ylabel = r"<$\sigma_z$>"
+    title = "Rabi Width Fit"
+
+    @staticmethod
+    def _model(x, *p):
+        return p[0] + p[1]*np.exp(-x/p[2])*np.cos(2*np.pi*p[3]*(x - p[4]))
+
+    def _initial_guess(self):
+        frabi, Tcs, amps = KT_estimation(self.ypts-np.mean(self.ypts), self.xpts, 1)
+        offset = np.average(self.xpts)
+        amp = np.max(self.ypts)
+        trabi = self.xpts[np.size(self.ypts) // 3]# assume Trabi is 1/3 of the scan
+        phase = 90.0
+        return [offset, amp, trabi, frabi[0], phase]
+
+    def _fit_dict(self, p):
+
+        return {"y0": p[0],
+                "A": p[1],
+                'T': p[2],
+                "f": p[3],
+                "phi": p[4]}
+
+    def __str__(self):
+        return "y0 + A*exp(-x/T)*cos(2*pi*f*(t - phi))"
+
+    @property
+    def t_rabi(self):
+        return self.fit_params["T"]
+
+    def annotation(self):
+        return r"$T_\pi$ = {0:.2e} {1} {2:.2e}".format(self.fit_params["T"], chr(177), self.fit_errors["T"])
+
+class T1Fit(AuspexFit):
+    """Fit to a decaying exponential for T1 measurement experiments.
+    """
+    xlabel = "Delay"
+    ylabel = r"<$\sigma_z$>"
+    title = r"$T_1$ Fit"
+
+    @staticmethod
+    def _model(x, *p):
+        return p[0]*np.exp(-x/p[1]) + p[2]
+
+    def _initial_guess(self):
```

Comment:
Random aside : the method [here](https://www.scribd.com/doc/14674814/Regressions-et-equations-integrales) (pg. 17) is super slick for getting the exponential decay guess. As a bonus @gribeill  should be able to read it :grinning:.

Reply:
Oh that looks like a very interesting reference... merci @caryan !